emcc.py

```diff
@@ -610,6 +610,11 @@ def check_human_readable_list(items):
   # previously used.
   if optimizing and not settings.SIDE_MODULE:
     passes += ['--zero-filled-memory']
+  # LLVM output always has immutable initial table contents: the table is
+  # fixed and may only be appended to at runtime. if the wasm is relocatable
+  # though then the offsets are dynamic anyhow, we can't actually optimize.
+  if optimizing and not settings.RELOCATABLE:
```

Comment:
I'm not sure that comment is correct.   In relocatable mode the segments are offset to `__table_base` but the segment contents are not mutable.. that are still immutable.

IIUC the directize pass already can't handle this case, but in theory it could be make to handle it.  In any case, I don't think it should be necessary to explictly opt out here in RELOCATABLE mode.

Reply:
Sounds good, I rewrote the comment and removed the relocatable check.